cwl-scripts/ffmpeg-docker-test/ffmpeg_trim.cwl

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+# uncomment next line if using docker image without ffmpeg as entry point
+# baseCommand: ffmpeg
+
+doc: trim mp4 file (from URL) and save it as <out_file_name>.mp4
+
+arguments: [
+  "-i", $(inputs.input_file_link), 
+  "-t", $(inputs.trim_seconds), 
+  "$(inputs.out_file_name).mp4"
+]
+
+hints:
+  DockerRequirement:
+    # self-created docker image
+    # dockerImageId: ffmpeg4
+
+    # public docker image
+    dockerPull: "jrottenberg/ffmpeg:4.0-ubuntu"
+
+inputs:
+  input_file_link: string
+  trim_seconds: int
+  out_file_name: string
+
+outputs:
+  out_video:
+    type: File
+    outputBinding:
+      glob: "$(inputs.out_file_name).mp4"
```

Comment:
how to get the output file out of container? will dockerOutputDirectory requirement help?

Reply:
Cwl will do this job for us automatically.

Reply:
[https://www.commonwl.org/user_guide/07-containers/index.html](https://www.commonwl.org/user_guide/07-containers/index.html) It connects docker's workdir with cwl tmp dir, so outputs are available outside the container. And files specified in outputs array are copied to cwl's workdir from tmp.

Reply:
<img width="624" alt="2018-11-05_20-58-37" src="https://user-images.githubusercontent.com/4573714/48016874-db1a7900-e13d-11e8-8cfa-eb6320e3f367.png">

<img width="411" alt="2018-11-05_21-01-51" src="https://user-images.githubusercontent.com/4573714/48016953-0dc47180-e13e-11e8-85c1-39de5d3fa3b0.png">